### PR TITLE
Rename Model -> DemographicModel.

### PR DESCRIPTION
**Pull request by grahamgower**
```
find stdpopsim tests qc docs -type f \
	-exec sed -i 's/\<Model\>/DemographicModel/g' {} \;
```
Plus line length fixes.

See #303.

### CODE REVIEW

File: stdpopsim/catalog/arabidopsis_thaliana.py [hunk 1]
```diff
@@ -118,11 +118,11 @@ def _sma_1pop():
             description="Arabidopsis Thaliana South Middle Atlas population")
     ]
 
-    return stdpopsim.Model(
+    return stdpopsim.DemographicModel(
         id="SMA_1pop",
         name="South Middle Atlas population size history",
         description="""
-            Model estimated from two homozygous individuals from
+            DemographicModel estimated from two homozygous individuals from
```
Comment (jeromekelleher) [inline stdpopsim/catalog/arabidopsis_thaliana.py]: This is a bit overzealous from sed --- I think the original English description is better.

Comment (jeromekelleher) [reply stdpopsim/catalog/arabidopsis_thaliana.py]: Ditto for various replacements of the word "Model" in English sentences below.

File: stdpopsim/engines.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def simulate(self, model=None, contig=None, samples=None, **kwargs):
         Simulates the model for the specified contig and samples.
 
         :param model: The demographic model to simulate.
```
Comment (jeromekelleher) [inline stdpopsim/engines.py]: Since this is a named parameter, should we change it to ``demographic_model``?

**Comment by jeromekelleher** [comment]
Thanks for this @grahamgower, I think we need to make this change before we ship and have to worry about backward compatability.

@andrewkern, any objections here? This will probably affect the analysis repo, but I can follow up there and make the necessary changes.

**Comment by andrewkern** [comment]
no objections to this. Jerome I will work through the analysis side today with the new refactors. @grahamgower thanks again for all this great work!

**Comment by jeromekelleher** [comment]
We also need ``Species.get_demographic_model`` @grahamgower

**Comment by jeromekelleher** [comment]
I had a quick look at the analysis repo @andrewkern --- looks like you'll need change change ``get_model`` to ``get_demographic_model`` and that should be about it (besides the demography debugger thing we talked about elsewhere)

**Comment by jeromekelleher** [comment]
Probably need to change the ``models`` attribute to ``demographic_models`` too, plus update the code for generating the catalog @grahamgower

**Comment by grahamgower** [comment]
Ok, I think I got most of them.

```
find docs stdpopsim tests qc -type f -exec sed -i 's/\<get_model\>/get_demographic_model/g' {} \;
find docs stdpopsim tests qc -type f -exec sed -i 's/\<add_model\>/add_demographic_model/g' {} \;
find docs stdpopsim tests qc -type f -exec sed -i 's/\<all_models\>/all_demographic_models/g' {} \;
find docs stdpopsim tests qc -type f -exec sed -i 's/\<species.models\>/species.demographic_models/g' {} \;
```

Plus a few manual changes.

**Comment by jeromekelleher** [comment]
Awesome, thanks @grahamgower! I've added a commit that changes a few DemographicModel cases to Demographic model in the docs. I'll merge once CI finishes.

**Pull request merged**